[Validator] Document constraints as php 8 Attributes (#14544)
symfony/symfony-docs: reference/constraints/Choice.rst

            protected $genre;
        }

    .. code-block:: php-attributes

        // src/Entity/Author.php
        namespace App\Entity;

        use Symfony\Component\Validator\Constraints as Assert;

        class Author
        {
            const GENRES = ['fiction', 'non-fiction'];

            #[Assert\Choice(["New York", "Berlin", "Tokyo"])]
            protected $city;

>             /**
>              * You can also directly provide an array constant to the "choices" option in the annotation
>              */
>             #[Assert\Choice(choices: Conference::GENRES, message: "Choose a valid genre.")]

Reviewer: ```suggestion
            // You can also directly provide an array constant to the "choices" option in the annotation.
            #[Assert\Choice(choices: Author::GENRES, message: 'Choose a valid genre.')]
```

The code comment explains the following attribute block. I find it confusing to use a phpdoc block for that.

Author: Just copy, paste from annotation example. Fixed

            protected $genre;
        }

    .. code-block:: php-attributes

        // src/Entity/Author.php
        namespace App\Entity;

        use Symfony\Component\Validator\Constraints as Assert;

        class Author
        {

>             #[Assert\Choice(callback: "getGenres")]

Reviewer: ```suggestion
            #[Assert\Choice(callback: 'getGenres')]
```

Author: Fixed

            protected $genre;
        }

    .. code-block:: php-attributes

        // src/Entity/Author.php
        namespace App\Entity;

        use Symfony\Component\Validator\Constraints as Assert;

        class Author
        {

>             #[Assert\Choice(callback: ["App\Entity\Genre", "getGenres"])]

Reviewer: ```suggestion
            #[Assert\Choice(callback: [Genre::class, 'getGenres'])]
```

Author: Fixed
